modules/gdscript/gdscript_parser.cpp

```diff
@@ -3041,6 +3048,41 @@ GDScriptParser::ExpressionNode *GDScriptParser::parse_dictionary(ExpressionNode
 	return dictionary;
 }
 
+GDScriptParser::ExpressionNode *GDScriptParser::parse_set(ExpressionNode *p_previous_operand, bool p_can_assign) {
+	DictionaryNode *dictionary = (DictionaryNode *)p_previous_operand;
```

Comment:
This makes the unchecked assumption that the passed operand is always a dictionary node. If this is intended make the type of the parameter `DictionaryNode *` instead to get compiler errors if used wrong. If not you will need to check whether the operand is a dictionary node by using `Node.type`